### ISSUE TITLE
v4.0.x: `--enable-install-libpmix` in a tarball does not work

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#6228]
While doing some testing for #6227, I noticed that `./configure --enable-install-libpmix` does not work from a tarball created from both master and the v4.0.x branch (I did not test farther back than this).  

Specifically, when you `make`, you get:

```
...
make[4]: Entering directory `/home/jsquyres/git/ompi/openmpi-gitclone/opal/mca/pmix/pmix4x/pmix/man'
../contrib/md2nroff.pl --source=pmix_init.3.md;
../contrib/md2nroff.pl --source=pmix_finalize.3.md;
../contrib/md2nroff.pl --source=pmix_initialized.3.md;
../contrib/md2nroff.pl --source=pmix_abort.3.md;
../contrib/md2nroff.pl --source=pmix_put.3.md;
../contrib/md2nroff.pl --source=pmix_commit.3.md;
../contrib/md2nroff.pl --source=pmix.7.md;
../contrib/md2nroff.pl --source=pmix_constants.7.md;
Source file does not exist (pmix_init.3.md) at ../contrib/md2nroff.pl line 38.
Source file does not exist (pmix_initialized.3.md) at ../contrib/md2nroff.pl line 38.
Source file does not exist (pmix_finalize.3.md) at ../contrib/md2nroff.pl line 38.
make[4]: *** [man3/pmix_init.3] Error 2
...
```

I note that this option works when you use it in a git clone (i.e., not a tarball build).  So it's probably something missing from `make dist`.

@rhc54 Is it expected that this CLI option is supposed to work?  If so, we should fix it. Otherwise, we should probably remove this CLI option.

FYI @georgemarselis

**Comment by ggouaillardet** [comment]
I will refresh the embedded `pmix4x` once pmix/pmix#1014 is merged

**Comment by ggouaillardet** [comment]
@jsquyres the PMIx refresh should have fixed the issue on `master`

**Comment by jsquyres** [comment]
I confirm that it's fixed on master.  Thanks!

Will you PR a fix to the v4.0.x branch?

**Comment by ggouaillardet** [comment]
@jsquyres I will do something, but I am not quite sure of what and when.

Ideally, this will be fixed in PMIx 3.0.3, and I will simply refresh it before Open MPI 4.0.1 is released.
If not, I found this simply trick is a good workaround for the `v4.0.x` branch

```patch
diff --git a/opal/mca/pmix/pmix3x/configure.m4 b/opal/mca/pmix/pmix3x/configure.m4
index 6188960eb3..f794412471 100644
--- a/opal/mca/pmix/pmix3x/configure.m4
+++ b/opal/mca/pmix/pmix3x/configure.m4
@@ -61,7 +61,8 @@ AC_DEFUN([MCA_opal_pmix_pmix3x_CONFIG],[
            CFLAGS="$OPAL_CFLAGS_BEFORE_PICKY $OPAL_VISIBILITY_CFLAGS"])
     AC_MSG_CHECKING([if want to install standalone libpmix])
     AS_IF([test "$enable_install_libpmix" == "yes"],
-          [AC_MSG_RESULT([yes])],
+          [AC_MSG_RESULT([yes])
+           opal_pmix_pmix3x_args="$opal_pmix_pmix3x_args ac_cv_prog_pmix_have_pandoc=no"],
           [AC_MSG_RESULT([no])
            opal_pmix_pmix3x_args="--with-pmix-symbol-rename=OPAL_MCA_PMIX3X_ --enable-embedded-mode $opal_pmix_pmix3x_args"])
     AS_IF([test "$with_devel_headers" = "yes"],
```

@rhc54 how should this be fixed in PMIx 3.0.3 ? remove the man pages ? I can understand this is not an acceptable option in the middle of a series, so should the `.md` files be added into the dist tarball ? I can issue a PR once you tell me which direction should be followed.

**Comment by rhc54** [comment]
It would probably be best to update to PMIx v3.1.0 - I doubt we will release a PMIx v3.0.3. I have posted a complete rc2 here: https://github.com/pmix/pmix/releases/tag/v3.1.0rc2. I expect we will do the official release by end of the week.

**Comment by hppritcha** [comment]
@gpaulsen I don't have an issue with updating to 3.1.0 do you?

**Comment by gpaulsen** [comment]
We'd love PMIx 3.1.0 in OMPI v4.0.x

**Comment by hppritcha** [comment]
@ggouaillardet we decided at today's devel meeting to refresh 4.0.x to PMIx 3.1.0 before releasing OMPI 4.0.1 so if you could add in the glue for that PMIx release once @rhc54 has released 3.1.0 that would be wonderful.

**Comment by gpaulsen** [comment]
@jsquyres Can you please reproduce with latest v4.0.x branch?

**Comment by gpaulsen** [comment]
@jsquyres Can we close this now?

**Comment by jsquyres** [comment]
New testing:

* With openmpi-master-201907130242-020a591: `--enable-install-libpmix` works fine.
* With openmpi-v4.0.x-201907100241-d3a7360: `--enable-install-libpimx` fails in configure with the following:

```
shell$ ./configure --prefix=$bogus --enable-install-libpmix
...lots of output...
--- MCA component pmix:pmix3x (m4 configuration macro)
checking for MCA component pmix:pmix3x compile mode... dso
checking if PMIx timing is enabled... no (disabled)
checking if want to install standalone libpmix... yes
configure: OPAL configuring in opal/mca/pmix/pmix3x/pmix
configure: running /bin/sh './configure' --disable-debug --disable-pmix-timing --without-tests-examples --disable-pmix-binaries --disable-pmix-backward-compatibility --disable-visibility --enable-embedded-libevent --with-libevent-header=\"opal/mca/event/libevent2022/libevent2022.h\" --enable-embedded-hwloc --with-hwloc-header=\"opal/mca/hwloc/hwloc201/hwloc201.h\"  '--prefix=/home/jsquyres/bogus' '--enable-install-libpmix' 'CFLAGS=-O3 -DNDEBUG  ' 'CPPFLAGS=-I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360 -I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360 -I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/include -I/home/jqsuyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/include -I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/mca/event/libevent2022/libevent -I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/mca/event/libevent2022/libevent/include -I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/mca/hwloc/hwloc201/hwloc/include  ' --cache-file=/dev/null --srcdir=. --disable-option-checking
configure: builddir: /home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/mca/pmix/pmix3x/pmix
configure: srcdir: /home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/mca/pmix/pmix3x/pmix

============================================================================
== Configuring PMIx
============================================================================
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu

...lots more output...

============================================================================
== Libevent
============================================================================
checking for libevent in... (default search paths)
looking for header in /usr/include
checking event.h usability... no
checking event.h presence... no
checking for event.h... no
looking for header in /usr/include/include
checking event.h usability... no
checking event.h presence... no
checking for event.h... no
configure: WARNING: LIBEVENT SUPPORT NOT FOUND
configure: error: CANNOT CONTINUE
configure: /bin/sh './configure' *failed* for opal/mca/pmix/pmix3x/pmix
checking if v3.x component is to be used... yes - using the internal v3.x library
configure: WARNING: INTERNAL PMIX FAILED TO CONFIGURE
configure: error: CANNOT CONTINUE
```

That is: it fails in the libevent configury inside the pmix configury inside the pmix:pmix3x configury.

**Comment by rhc54** [comment]
Well, duh - that configure option doesn't exist in PMIx v3.x! So I should hope it would fail.

**Comment by rhc54** [comment]
Oh, I see - it is an OPAL level configure item. Never mind - not sure what the problem is here but it is in that OPAL configure code.

**Comment by jsquyres** [comment]
Oddly enough:

* It fails in Open MPI v4.0.0 with the error in the original description
* It *works* with Open MPI v4.0.1! 😱
* It fails with Open MPI v4.0.x HEAD (e.g., openmpi-v4.0.x-201907100241-d3a7360), with a different error than the originally-reported error (see https://github.com/open-mpi/ompi/issues/6228#issuecomment-511567413)

**Comment by rhc54** [comment]
You know, to be fair, I know why we added that option - but the reality is that the packagers have now picked PMIx up and there is no good reason to be emitting that library...and many good reasons NOT to do so. I would recommend just removing the option.

**Comment by rhc54** [comment]
I just tested the head of the v4.0.x branch and it configures and builds just fine with `--enable-install-libpmix` on my Mac. However, I still strongly advise we remove that option.

**Comment by hppritcha** [comment]
Probably will not fix in 4.0.x release stream.  Moving to 4.1.0 milestone.

**Comment by jsquyres** [comment]
> I just tested the head of the v4.0.x branch and it configures and builds just fine with `--enable-install-libpmix` on my Mac. However, I still strongly advise we remove that option.

FWIW, I confirm that this option does not exist on master / will not be in Open MPI v5.x.

I'm hesitant to remove it in Open MPI v4.x because of backwards compatibility issues (for at least the environments where it does currently work).

**Comment by jsquyres** [comment]
I take back my prior comment from a few minutes ago, because:

* `--enable-install-libpmix` (apparently) works on a git clone (according to past Jeff, who initially filed this issue 😄)
* It does _not_ work on any Open MPI v4.0.x tarball (I just manually tested this)

Hence, it has never worked for users.  Hence, there's no backwards compatibility issue.

Hence, I agree with Ralph: let's take it out of the v4.0.x and v4.1.x available configure options.

**Comment by jsquyres** [comment]
This was merged in both the v4.0.x and v4.1.x branches.  Closing the issue.
